runtime/src/identity.rs

```diff
@@ -380,6 +396,136 @@ decl_module! {
             Ok(())
         }
 
+        /// Initiates master key rotation for a DID.
+        ///
+        /// # Failure
+        /// Only called by master key owner.
+        ///
+        /// # Arguments
+        /// * `did` caller's DID
+        /// * `new_key` master key replacing current key
+        pub fn change_master_key(origin, did: IdentityId, new_key: Key) -> Result {
+            let sender = ensure_signed(origin)?;
+            let master_key = Key::try_from(sender.encode())?;
+            let _grants_checked = Self::grant_check_only_master_key(&master_key, did)?;
+
+            ensure!(!<MasterKeyRotation>::exists(did), "a key rotation process already in progress");
```

Comment:
Maybe this should just invalidate the previous MasterKeyRotation and create a new one?

Reply:
Looking at the code, it looks like if I create a new MasterKeyRotation, then I am stuck until the MKR is completed (i.e. newKey and KYC have approved it). I think we also need the ability for the current master key to invalidate the rotation (assuming it hasn't already been approved by newKey & KYC SP).

Also, on a related note, wondering what would happen if the current master key invalidated the authorisation. I think this wouldn't change the MasterKeyRotation - would that be a recoverable state?

Reply:
I wanted to be explicit and throw an error if there's a duplicate request. Regarding backing out of a key rotation process. we have two choices:
1. Set an expiry on auth so request times out and we remove the `MasterKeyRotation` request. After which user makes a new request.
2. Provide an explicit method to cancel a request.

I think #1 is better.

If the auth is invalidated, we'd reset the state. `MasterKeyRotation` is removed.